add comparison/plotting code for original vs new compendia

taylorreiter opened the pull request:
This PR adds two notebooks to this repository. They analyze the outputs of the snakefile `add_sra_to_compendia.snakefile`.

+ `[notebooks/20220315-compare-compendia.ipynb](https://github.com/taylorreiter/2022-cf-sputum/blob/comp_comp/notebooks/20220315-compare-compendia.ipynb)`: compares the original to the new compendia. The new compendia contain 10 sputum samples that were not included in the original compendia. The goal of the notebook was to determine whether the filtering and normalization behaved similarly, even when 10 new samples were added. Performs linear regression of a single sample, a mantel test between the two compendia, and PCA to see if similar results are produced.
+ `[notebooks/20220316_compare_raw_counts_values.ipynb](https://github.com/taylorreiter/2022-cf-sputum/blob/comp_comp/notebooks/20220316_compare_raw_counts_values.ipynb)`: Because some samples are filtered out of the compendia for not having enough reads, I processed some samples that were in the _raw_ compendia but not in the _filtered_ compendia. There were five such samples, and these samples allowed me to assess whether just the salmon counting step produced the same results when @georgiadoing performed counting vs. when I performed counting.

taylorreiter committed:
add comparison/plotting code for og vs new compendia

ajlee21 reviewed (approved):
Nice analysis of things not changing after you add new data. Just had a few questions about the figures and what they mean.

Comments for `20220315-compare-compendia.ipynb`
1. Very nice description in the headers!
2. I've used Rmagic to run some cells in R, but looks like you're using an R kernel here. Neat!
3. I very much agree with Casey's point about the outliers really affecting the correlation's you'll see so the scaling makes sense. I did also do a scaling for the core-acc paper as well.
4. Might be helpful to explain what the PC plots are showing. I think without the meeting, I wouldn't know what is being compared. Also, I see for the last PC plot you have PC1 compared to PC1.1 but for the other PC plot you don't appear to have the same mapping which confused me a bit

Comments for `20220316_compare_raw_counts_values.ipynb`
1. Sorry I'm a little lost at what this correlation is showing. So you're comparing the raw reads mapped to sample SRX4632312 using the original compendium vs using the new compendium? But I don't know how the different compendium will change the raw reads you find?

taylorreiter committed:
switch corrgram to individual linear regressions and plots

taylorreiter committed:
add context to explain confusing pca labels

taylorreiter commented:
> Nice analysis of things not changing after you add new data. Just had a few questions about the figures and what they mean.
> 
> Comments for `20220315-compare-compendia.ipynb`
> 
>     4. Might be helpful to explain what the PC plots are showing. I think without the meeting, I wouldn't know what is being compared. Also, I see for the last PC plot you have PC1 compared to PC1.1 but for the other PC plot you don't appear to have the same mapping which confused me a bit
> 
Thanks this is helpful feedback. I added descriptions to these plots.
I don't think I understand the second part of your comment though. Could you explain it in a different way? I think the mapping is the same between both PCs, but I might be missing something.
> 
> Comments for `20220316_compare_raw_counts_values.ipynb`
> 
>     1. Sorry I'm a little lost at what this correlation is showing. So you're comparing the raw reads mapped to sample SRX4632312 using the original compendium vs using the new compendium? But I don't know how the different compendium will change the raw reads you find?

Thank you for this question. I changed the plots to make it more clear what I was trying to show, and added more context. The purpose was to show that for the 5 samples that I accidentally reprocessed (that Georgia also processed, but that got filtered out and didn't make it into the compendia), that we get the ~same counts as Georgia got. I didn't control which version of salmon I used compared to what georgia used, so there's a chance I could have produced drastically different counts. I wanted to make sure that we didn't, so just a quality check.

ajlee21 commented:
Sorry I think I mixed up some of the numbering in the PC's when I looked at it. The mapping looks the same to me now. I think the added comments really help. Nice job!
